### DISCUSSION TITLE
Does langium support generator generation? (Xtend in Xtext)?

### DISCUSSION THREAD

**Discussion by yylonly** [eclipse-langium/langium#736] [Q&A] [answered]

**Comment by msujew** [comment, marked as answer]
Hey @yylonly,

we don't offer a new language for generation, but instead just a relatively simple [generator API](https://github.com/langium/langium/blob/main/packages/generator-langium/langium-template/src/cli/generator.ts) to perform code generation. An example is contained in the yeoman generator for Langium.

Please note that the JS ecosystem also offers tons of other templating engines such as [doT](https://github.com/olado/doT) and others. So you can use those as well if you don't like the minimalist API that we provide.
